src/core_modules/capture-core/components/ListView/types/listView.types.ts

```diff
@@ -0,0 +1,129 @@
+import type { ReactNode, ReactElement } from 'react';
+import type { dataElementTypes } from '../../../metaData';
+import type { FilterData, Options } from '../../FiltersForTypes';
+
+export type Column = {
+    id: string;
+    visible: boolean;
+    type: dataElementTypes[keyof dataElementTypes];
+    header: string;
+    options?: Options;
+    multiValueFilter?: boolean;
+    filterHidden?: boolean;
+    additionalColumn?: boolean;
+    unique?: boolean;
+};
+
+export type Columns = Array<Column>;
+
+export type FilterOnly = {
+    id: string;
+    type: dataElementTypes[keyof dataElementTypes];
+    header: string;
+    options?: Options;
+    multiValueFilter?: boolean;
+    tooltipContent?: string;
+    disabled?: boolean;
+    showInMoreFilters?: boolean;
+    transformRecordsFilter: (rawFilter: any) => Record<string, unknown>;
+};
+
+export type AdditionalFilter = FilterOnly & {
+    mainButton?: boolean;
+};
+
+export type FiltersOnly = Array<FilterOnly>;
+export type AdditionalFilters = Array<AdditionalFilter>;
+export type DataSourceItem = { [id: string]: string };
+
+export type DataSource = Array<DataSourceItem>;
+
+type CustomMenuContentItem = {
+    key: string;
+    clickHandler?: () => void;
+    element: ReactNode;
+};
+
+type CustomMenuContentHeader = {
+    key: string;
+    subHeader: ReactNode;
+};
+
+export type CustomMenuContent = CustomMenuContentItem | CustomMenuContentHeader;
+
+export type CustomMenuContents = Array<CustomMenuContent>;
+
+export type CustomRowMenuContent = {
+    key: string;
+    clickHandler?: (rowData: DataSourceItem) => any;
+    icon?: ReactNode;
+    label: string;
+};
+export type CustomRowMenuContents = Array<CustomRowMenuContent>;
+
+export type CustomTopBarActions = Array<{key: string, actionContents: ReactNode}>;
+
+export type FiltersData = { [id: string]: FilterData };
+
+export type PaginationContextData = {
+    onChangePage: (pageNumber: number) => void;
+    onChangeRowsPerPage: (rowsPerPage: number) => void;
+    rowsPerPage: number;
+    currentPage: number;
+    rowCountPage: number;
+};
+
+export type StickyFilters = {
+    userSelectedFilters?: { [id: string]: boolean };
+    filtersWithValueOnInit?: { [id: string]: boolean };
+};
+
+export type ChangePage = (pageNumber: number) => void;
+export type ChangeRowsPerPage = (rowsPerPage: number) => void;
+export type UpdateFilter = (id: string, data?: Record<string, unknown>) => void;
```

Comment:
Revert the order of the args to `export type UpdateFilter = (data: Record<string, unknown>, id: string) => void;`